File: code/Java/interpreter/command/BlocksCommand.java
package interpreter.command;

import java.util.ArrayList;
import java.util.List;

public class BlocksCommand extends Command {

    private List<Command> cmds;

    public BlocksCommand(int line) {
        super(line);
        cmds = new ArrayList<Command>();
    }

    public void addCommand(Command cmd) {
        cmds.add(cmd);
    }

    public void execute() {
        for (Command cmd : cmds)
            cmd.execute();
    }

}

File: code/C++/interpreter/command/BlocksCommand.cpp
#include "BlocksCommand.h"

BlocksCommand::BlocksCommand(int line)
	: Command(line) {
}

BlocksCommand::~BlocksCommand() {
	for (std::list<Command*>::iterator it = m_cmds.begin(),
	     ed = m_cmds.end(); it != ed; it++) {
		Command* cmd = *it;
		delete cmd;
	}
}

void BlocksCommand::addCommand(Command* cmd) {
	m_cmds.push_back(cmd);
}

void BlocksCommand::execute() {
	for (std::list<Command*>::iterator it = m_cmds.begin(),
	     ed = m_cmds.end(); it != ed; it++) {
		Command* cmd = *it;
		cmd->execute();
	}
}
